frontend/packages/console-dynamic-plugin-sdk/src/api/internal-api.ts

```diff
@@ -53,6 +55,15 @@ export const VirtualizedGrid: React.FC<VirtualizedGridProps> = require('@console
 export const LazyActionMenu: React.FC<LazyActionMenuProps> = require('@console/shared/src/components/actions/LazyActionMenu')
   .default;
 
+export const DroppableEditYAML: React.FC<DroppableEditYAMLProps> = require('@console/internal/components/droppable-edit-yaml')
+  .DroppableEditYAML;
+
+export const LabelsModal: React.FC = require('@console/internal/components/modals/labels-modal')
+  .labelsModal;
```

Comment:
AFAIK the `labelsModal` here (and `annotationsModal` below) are modal creator functions, which is why their name starts with a lowercase letter. Both `labelsModal` and `annotationsModal` delegate to `createModalLauncher` function.

TS tells me that `annotationsModal` function has the following signature:
```ts
(props: AnnotationsModalProps & CreateModalLauncherProps) => { result: Promise<{}> }
```

To have proper typings, I'd suggest the following:

1) update types in `public/components/factory/modal.tsx`

```ts
export type CreateModalLauncherResult<P extends ModalComponentProps> = (
  props: P & CreateModalLauncherProps,
) => { result: Promise<{}> };

export type CreateModalLauncher = <P extends ModalComponentProps>(
  C: React.ComponentType<P>,
) => CreateModalLauncherResult<P>;
```

2) update code in `packages/console-dynamic-plugin-sdk/src/api/internal-api.ts`

```ts
export const annotationsModal: CreateModalLauncherResult<AnnotationsModalProps> = require('@console/internal/components/modals/tags')
  .annotationsModal;
```

And also the same thing for `labelsModal` above.

This way, we shouldn't need to do any changes in files that contain `labelsModal` and `annotationsModal` functions.